MAINTAINERS.md

```diff
@@ -77,8 +87,73 @@ Make frequent project releases to the community.
+
+### Public Confirmation
+
+Upon receiving 3 positive maintainer votes, and no vetoes, from other maintainers, and after having confirmed interest with the nominee, the maintainer opens a public issue in order to make the vote visible to the community. 
+
+> _Content from the above nomination._
+> 
+> Here is my +1.
+> 
+> If you support this nomination, please add your +1 or a comment. 
+> 
+> If you do not support this nomination, we invite you to e-mail all/some of the current [project maintainers](https://github.com/opensearch-project/OpenSearch/blob/main/MAINTAINERS.md) in the next 2 weeks. We will keep your feedback private to this group and discuss offline. We will not publicly post the reasons for a decision not to add this co-maintainer, or publicize your name. 
+> 
+> Anyone, including maintainers, can publicly add +1s as requested in the nomination issue if they support the nomination or contact the maintainers in private if they oppose. The issue stays open for 2 weeks.
+
+### Maintainer Decision
+
+If a positive decision has been reached, the nomination is accepted and the nominating maintainer can thus comment on the issue with _"maintainers has been added"_. If a decision cannot be reached for any reason, the nomination is rejected with _"a decision could not be reached"_ with no further explanation necessary.
+
+### Adding the New Maintainer 
+
+The Repo Admin adjusts the new maintainer’s permissions accordingly. The nominating maintainer opens a pull request to add the new maintainer to MAINTAINERS.md, and closes the nomination issue when that is merged and the individual has been given maintainer-level permissions.
+
+## Removing a Maintainer
+
+Removing a maintainer is a disruptive action that the community of maintainers should not undertake lightly. There are several reasons a maintainer might be removed from the project, such as inactivity, or taking actions that negatively impact the project.
+
+### Moving On
+
+There are plenty of reasons that might cause someone to want to take a step back or even a hiatus from a project. Existing maintainers can choose to leave the project at any time, with or without reason, by making a pull request to move themselves to the "Emeritus" section of MAINTAINERS.md, and asking an admin to remove their permissions.
+
+### Inactivity
+
+If the maintainers on the project notice that another maintainer is no longer an active or visible member of the community, the first step will be to tell the repo owner to contact the inactive maintainer to check-in. If the inactive maintainer responds and intends to reengage, no action is required. If the maintainer does not respond after a week, or indicates they no longer wish to be a maintainer, the repo owner can remove the inactive maintainer’s permissions and make a pull request to add them to the "Emeritus" section of the MAINTAINERS.md.
+
+### Negative Impact on the Project
```

Comment:
Would not meeting the responsibilities outlined in the `MAINTAINERS.md` document fall into this section and constitute basis for removal too?

A separate conversation is whether riding on the coattails of this initiative we want to also revisit that document to make sure it still represents the bar we hold our maintainers to

Reply:
I think almost all of that would fall under not being active/visible.  Did you have a specific thing in mind, Dave?

It's part of the same doc, so we can always take a look at it :)

Reply:
In general, in open-source one cannot expect or ask for time/work/engagement from others, so I'd be careful writing anything about maintainer responsibilities being a job.

Reply:
I think those responsibilities are "responsibilities are tenets" rather than "responsibilities are a job". A maintainer cannot be expected, for example, to respond to an issue in 24 hours, but can be expected to be responsive to issues. Does this help? Am I thinking about it wrong? Either way, if you think there are words that would clarify this in the current change to MAINTAINERS.md, I'll gladly consider!

Reply:
Makes sense. On second review of the maintainers file it phrases them with terms like "regularly" instead of being prescriptive, so I don't think any changes are needed. Thanks!

Reply:
I would suggest changing 'Actions' to 'Actions (or inactions)' which covers some of my concerns about inactivity.

Reply:
I think I disagree. Inactions rarely have negative impact as meant by this paragraph (code of conduct violations, security risks). We talk a lot about inactivity above to avoid associating the two.

Reply:
@dblock Oh there are lots of ways inactions by maintainers negatively impact the project. Off the top of my head:
- Security: Failing to merge a security fix
- CoC: Failing to flag violations

Reply:
Are maintainers alone expected to flag CoC violations? This is a community wide responsibility and i feel like maintainers are not in this alone. Its the community + the **admins** that are responsible for this imo.

Reply:
Certainly not, but this is a MAINTAINERS doc. In the CoC we have language that talks about reporting these. Speaking of which, https://github.com/opensearch-project/.github/pull/69.